docs/site/DataSource-generator.md

```diff
@@ -50,10 +50,93 @@ Once all the prompts have been answered, the CLI will do the following:
 
 - Install `@loopback/repository` and the connector package (if it's not a custom
   connector).
-- Create a file with the connector configuration as follows:
-  `/src/datasources/${dataSource.dataSourceName}.datasource.config.json`
-- Create a DataSource class which recieves the connector config using
+- Create a DataSource class which receives the connector config using
   [Dependency Injection](Dependency-injection.md) as follows:
-  `/src/datasources/${dataSource.dataSourceName}.datasource.ts`
+  `/src/datasources/${dataSourceName}.datasource.ts`. The connector
+  configuration provided via CLI is available via static property
+  `defaultConfig`.
 - Update `/src/datasources/index.ts` to export the newly created DataSource
   class.
+
+### Legacy JSON-based configuration
+
+Datasources created by a CLI version from before April 2020 are using JSON-based
+file convention for defining datasources:
+
+- `/src/datasources/${dataSourceName}.datasource.config.json` contains
+  datasource configuration in JSON format, e.g. the connector name, target host
+  & port, credentials, and so on.
+- `/src/datasources/${dataSourceName}.datasource.ts` defines a datasource class,
+  the default configuration is loaded from the JSON file.
+
+CLI commands like `lb4 repository` and `lb4 service` support both the old
+(JSON-based) and the current (pure TypeScript) styles, there are no changes
+immediately needed to make your existing project work with the recent CLI
+versions.
+
+However, we recommend to eventually migrate existing datasource files to the new
+style.
+
+#### Migration guide
+
+1. Open `{dataSourceName}.datasource.ts` file and replace import of `config`
+   from the JSON file with the actual content of the JSON file.
+
+   For example:
+
+   ```diff
+   -import config from './db.datasource.config.json';
+   +
+   +const config = {
+   +  "name": "db",
+   +  "connector": "memory",
+   +  "localStorage": "",
+   +  "file": "./data/db.json",
+   +};
+   ```
+
+2. Save the file, run Prettier to fix the coding style (change double-quotes `"`
+   to single-quotes `'`, remove quotes from property names like `connector` and
+   `file`).
+
+   Example output using LoopBack's coding style:
+
+   ```ts
+   const config = {
+     name: 'db',
+     connector: 'memory',
+     localStorage: '',
+     file: './data/db.json',
+   };
+   ```
+
+3. Add a static `defaultConfig` property to your datasource class.
+
+   For example:
+
+   ```diff
+   export class DsDataSource extends juggler.DataSource {
+     static dataSourceName = 'ds';
+   +  static readonly defaultConfig = config;
+
+     constructor(
+   ```
+
+4. Modify the tests importing the default datasource configuration from the JSON
+   file, get the default configuration via the new static property instead.
+
+   For example:
+
+   ```diff
+   -import GEO_CODER_CONFIG from '../datasources/geocoder.datasource.config.json';
+   +import {GeocoderDataSource} from '../datasources/geocoder.datasource';
```

Comment:
Maybe we could use the same datasource throughout instead of using all of `db`, `ds`, and the geocoder one

Reply:
Good catch. I'll unify `db` and `ds` into one name.

Geocoder is a bit special, it's a datasource connected to a webservice (not a database), our users may be familiar with this datasource from our Todo tutorial - see https://loopback.io/doc/en/lb4/todo-tutorial-geocoding-service.html I prefer to keep geocoder as is now.

Reply:
Fixed in 23c03c99.